src/transformers/utils/model_parallel_utils.py

```diff
@@ -12,10 +12,186 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import copy
+import importlib
+from collections import defaultdict
 from math import ceil
 
 
+class TPInfo(object):
+    def __init__(
+        self,
+        *name,
+        combined_qkv: bool = False,
+        reverse: bool = False,
+    ):
```

Comment:
```suggestion
    def __init__(self, *name, combined_qkv: bool = False, reverse: bool = False):
```
This fits in one line (our char limit is 119).